### PR TITLE
testing/bloaty: new aport

### PR DESCRIPTION
**Pull request by mitchty**
Note, this entire apkbuild is a hack, there is no upstream source
tarball anywhere. So I hacked in the build function to git clone
the source and checkout at a specific known version for now.

Not sure what the best option is in cases like these.

### CODE REVIEW

**Comment by fabled** [comment]
For github you can download tarball of specific commit. For an example, see: main/raspberrypi/APKBUILD.

**Comment by mitchty** [comment]
Ah cool, I keep forgetting about that feature of github, I'll update this to use that then thanks!

**Comment by mitchty** [comment]
So it turns out there were three submodules in this too, I couldn't find a release for one of the three, libFuzzer as used from chromium. So not sure how to handle that one outside of just checking it out at a certain point in time.

File: testing/bloaty/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#-*-mode: Shell-script; coding: utf-8;-*-
+# Maintainer: Mitch Tishmack <mitch.tishmack@gmail.com>
+_commitid=2234386bcee7297dfa1b6d8a5d20f95ea4ed9bb0
+pkgname=bloaty
+# Fake a version until upstream does one
+# format is simply 0.0.0.0.${commit_count}
+pkgver=0.0.0.0.79
```
Comment (jirutka) [inline testing/bloaty/APKBUILD]: Sigh, yet another Google project that doesn’t bother to do releases…

For such cases it’s IMO better to use version number in format `0.0_gitYYYYMMDD` (i.e. `0.0_git20161228`) or if you prefer commits count, then `0.0_gitX` (i.e. `0.0_git79`). It makes it more clear that it’s a snapshot from git.

File: testing/bloaty/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# Fake a version until upstream does one
+# format is simply 0.0.0.0.${commit_count}
+pkgver=0.0.0.0.79 
+pkgrel=0
+pkgdesc="A size profiler for binaries"
+url="https://github.com/google/bloaty"
+arch="all"
+license="apache"
+depends=""
+makedepends="$depends gcc git"
```
Comment (jirutka) [inline testing/bloaty/APKBUILD]: Don’t add `$depends` into `makedepends`, this is implicit.

File: testing/bloaty/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	byteswap.patch
+	"
+builddir="$srcdir/${pkgname}-${_commitid}"
+
+build() {
+	rmdir third_party/re2 third_party/googletest || return 1	
+	ln -s ../../re2-2017-01-01 third_party/re2 || return 1
+	ln -s ../../googletest-release-1.8.0 third_party/googletest || return 1
+	git clone https://chromium.googlesource.com/chromium/llvm-project/llvm/lib/Fuzzer third_party/libFuzzer || return 1
+	(cd third_party/libFuzzer && git checkout e9387f0b6d411e18ca75f08aecfec1238b8e113f) || return 1
```
Comment (jirutka) [inline testing/bloaty/APKBUILD]: Please move this SHA into a top-level variable, so one does not simply overlook that there’s some hard-coded version.

File: testing/bloaty/APKBUILD (hunk 1 above)
Comment (mitchty) [inline testing/bloaty/APKBUILD]: For the date should it be the last commit or just say today? Or is it more whatever works in this case, I'm impartial really, my goal with that version was to make a silly version that was almost guaranteed to be lower than whatever was chosen by the upstream.

File: testing/bloaty/APKBUILD (hunk 3 above)
Comment (mitchty) [inline testing/bloaty/APKBUILD]: Will do, also would the git checkout be better done in a prepare function?

File: testing/bloaty/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+	https://github.com/google/googletest/archive/release-1.8.0.tar.gz
+	byteswap.patch
+	"
+builddir="$srcdir/${pkgname}-${_commitid}"
+
+build() {
+	rmdir third_party/re2 third_party/googletest || return 1	
+	ln -s ../../re2-2017-01-01 third_party/re2 || return 1
+	ln -s ../../googletest-release-1.8.0 third_party/googletest || return 1
+	git clone https://chromium.googlesource.com/chromium/llvm-project/llvm/lib/Fuzzer third_party/libFuzzer || return 1
```
Comment (kaniini) [inline testing/bloaty/APKBUILD]: i don't like this, it should be possible to build a package without requiring a functioning internet connection.

if we have to git clone during the build process, then it makes the build process much more complex

File: testing/bloaty/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	https://github.com/google/re2/archive/2017-01-01.tar.gz
+	https://github.com/google/googletest/archive/release-1.8.0.tar.gz
+	byteswap.patch
+	"
+builddir="$srcdir/${pkgname}-${_commitid}"
+
+build() {
+	rmdir third_party/re2 third_party/googletest || return 1	
+	ln -s ../../re2-2017-01-01 third_party/re2 || return 1
+	ln -s ../../googletest-release-1.8.0 third_party/googletest || return 1
```
Comment (kaniini) [inline testing/bloaty/APKBUILD]: there is googletest here which indicates a testsuite is present; but no `check()` function.

Comment (mitchty) [inline testing/bloaty/APKBUILD]: It looks like that causes this:

```
git submodule init && git submodule update
make: *** No rule to make target 'third_party/googletest/googlemock/gtest/libgtest.a', needed by 'tests/range_map_test'.  Stop.
make: *** Waiting for unfinished jobs....                                                                                                                        touch .gitmodules -r third_party/googletest/CMakeLists.txt
>>> ERROR: bloaty: all failed
>>> bloaty: Uninstalling dependencies...
```

I'll try a prepare and just tar.gz a full git checkout, at this point the things this depends on are fractal in missing sub dependencies.

**Comment by mitchty** [comment]
Ok I gave up and created a snapshot function that just tar.xz's the entire git checkout with submodules and threw it up on my vps. I'm not sure thats the best way to do this or if it would better be on alpinelinux.org or not. Otherwise I'll just keep building this thing as a static binary and just leave this be.

Note I committed this via git commit --squash HEAD and didn't rebase --autosquash so the differences can be seen.

**Comment by mitchty** [comment]
Looks like the make test needs cmake, i'll fix it.

**Comment by mitchty** [comment]
ugh, I give up, I'll just keep building the binary statically.

**Pull request closed without merging**